pkg/util/util.go

```diff
@@ -108,3 +111,21 @@ func GetCanonicalTypeName(model interface{}) string {
 	}
 	return path + "." + t.Name()
 }
+
+// Provides a fast path for decoding YAML scalar node as a string
+// If the node's value can be simply returned directly, then it is. Otherwise,
+// the yaml.v3.Node.Decode slow path is taken
+func DecodeYAMLString(n *yaml.Node, s *string) error {
+	if n.Kind != yaml.ScalarNode {
+		return fmt.Errorf("expected scalarnode, not %v", n.Kind)
+	}
+
+	if (n.Tag == "!!str" || n.Tag == "tag:yaml.org,2002:!!string") ||
+		(n.Tag == "" || n.Tag == "!") && n.Style&(yaml.SingleQuotedStyle|yaml.DoubleQuotedStyle|yaml.LiteralStyle|yaml.FoldedStyle) != 0 {
```

Comment:
is this copied from something in the yaml library? I'm not familiar with what this means or if it is correct

Reply:
I copied it from yaml decoding logic for string path:

https://github.com/go-yaml/yaml/blob/496545a6307b2a7d7a710fd516e5e16e8ab62dbc/yaml.go#L463
https://github.com/go-yaml/yaml/blob/496545a6307b2a7d7a710fd516e5e16e8ab62dbc/decode.go#L565